substrate/frame/revive/src/vm/evm.rs

```diff
@@ -124,380 +128,16 @@ impl<T: Config> ContractBlob<T> {
 }
 
 /// Calls the EVM interpreter with the provided bytecode and inputs.
-pub fn call<'a, E: Ext>(bytecode: Bytecode, ext: &'a mut E, inputs: EVMInputs) -> ExecResult {
-	let mut interpreter: Interpreter<EVMInterpreter<'a, E>> = Interpreter {
-		gas: Gas::default(),
-		bytecode: ExtBytecode::new(bytecode),
-		stack: Stack::new(),
-		return_data: Default::default(),
-		memory: SharedMemory::new(),
-		input: inputs,
-		runtime_flag: RuntimeFlags { is_static: ext.is_read_only(), spec_id: SpecId::default() },
-		extend: ext,
-	};
-
-	let table = instruction_table::<'a, E>();
-	let result = run(&mut interpreter, &table);
-
-	instruction_result_into_exec_error::<E>(result.result)
-		.map(Err)
-		.unwrap_or_else(|| {
-			Ok(ExecReturnValue {
-				flags: if result.is_revert() { ReturnFlags::REVERT } else { ReturnFlags::empty() },
-				data: result.output.to_vec(),
-			})
-		})
+pub fn call<E: Ext>(bytecode: Bytecode, ext: &mut E, input: Vec<u8>) -> ExecResult {
+	let mut interpreter = Interpreter::new(ExtBytecode::new(bytecode), input, ext);
+	let ControlFlow::Break(halt) = run_plain(&mut interpreter);
+	interpreter.into_exec_result(halt)
 }
 
-/// Runs the EVM interpreter
-fn run<'a, E: Ext>(
-	interpreter: &mut Interpreter<EVMInterpreter<'a, E>>,
-	table: &revm::interpreter::InstructionTable<EVMInterpreter<'a, E>, DummyHost>,
-) -> InterpreterResult {
-	let host = &mut DummyHost {};
+fn run_plain<E: Ext>(interpreter: &mut Interpreter<E>) -> ControlFlow<Halt, Infallible> {
```

Comment:
note I will re-introduce the tracing version in a follow up PR that will add opcode tracing and let you compare opcode / stack and memory executed on revive vs revm eth interpreter